### ISSUE TITLE
Go SDK Wrapper stopped working

### ISSUE THREAD

**Issue by BillMerryman** [kercre123/wire-pod#118]
I had a plugin working with the SDK wrapper, but it has now stopped working. Seems like some kind of issue with getting the locale. I tried removing my plugin, and testing with the sdkTest.go file. Using it unchanged, it would throw an error. The sample file uses InitSDKForWirepod, which I have found does tend to throw errors. My working example was using InitSDK, and that WAS working. I updated the sdkTest.go file to use InitSDK, and it was throwing the same error as my previously working plugin. The error text is as follows:

Mar 23 13:29:13 escapepod start.sh[1401]: panic: interface conversion: interface {} is nil, not string
Mar 23 13:29:13 escapepod start.sh[1401]: goroutine 46 [running]:
Mar 23 13:29:13 escapepod start.sh[1401]: github.com/fforchino/vector-go-sdk/pkg/sdk-wrapper.GetLocale(...)
Mar 23 13:29:13 escapepod start.sh[1401]:         /root/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20221216080054-dfe1a258d335/pkg/sdk-wrapper/sdk-wrapper-settings.go:127
Mar 23 13:29:13 escapepod start.sh[1401]: github.com/fforchino/vector-go-sdk/pkg/sdk-wrapper.refreshLanguage()
Mar 23 13:29:13 escapepod start.sh[1401]:         /root/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20221216080054-dfe1a258d335/pkg/sdk-wrapper/sdk-wrapper-voice.go:147 +0xe2c
Mar 23 13:29:13 escapepod start.sh[1401]: github.com/fforchino/vector-go-sdk/pkg/sdk-wrapper.RefreshSDKSettings()
Mar 23 13:29:13 escapepod start.sh[1401]:         /root/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20221216080054-dfe1a258d335/pkg/sdk-wrapper/sdk-wrapper-settings.go:52 +0xc8
Mar 23 13:29:13 escapepod start.sh[1401]: github.com/fforchino/vector-go-sdk/pkg/sdk-wrapper.InitSDK({0x40005ae010?, 0x4000010018?})
Mar 23 13:29:13 escapepod start.sh[1401]:         /root/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20221216080054-dfe1a258d335/pkg/sdk-wrapper/sdk-wrapper.go:54 +0x4a4
Mar 23 13:29:13 escapepod start.sh[1401]: plugin/unnamed-2724764ea56c68990762bcca8dcd671b059c2a09.Action({0x143b680?, 0x4000010018?}, {0x40005ae010, 0x8})
Mar 23 13:29:13 escapepod start.sh[1401]:         /home/pi/wire-pod/chipper/plugins/sdkTest.go:18 +0x6c
Mar 23 13:29:13 escapepod start.sh[1401]: github.com/kercre123/chipper/pkg/wirepod/ttr.pluginFunctionHandler({0x1108160, 0x400036a7e0}, {0x40004901a0, 0xb}, 0x0?, {0x40005ae010, 0x8})
Mar 23 13:29:13 escapepod start.sh[1401]:         /home/pi/wire-pod/chipper/pkg/wirepod/ttr/matchIntentSend.go:179 +0x188
Mar 23 13:29:13 escapepod start.sh[1401]: github.com/kercre123/chipper/pkg/wirepod/ttr.ProcessTextAll({0x1108160?, 0x400036a7e0?}, {0x40004901a0?, 0x10?}, {0x400020e600, 0x36, 0x400?}, {0x4000065c00?, 0x36, 0x40}, ...)
Mar 23 13:29:13 escapepod start.sh[1401]:         /home/pi/wire-pod/chipper/pkg/wirepod/ttr/matchIntentSend.go:214 +0x12c
Mar 23 13:29:13 escapepod start.sh[1401]: github.com/kercre123/chipper/pkg/wirepod/preqs.(*Server).ProcessIntentGraph(0x1448220?, 0x400036a7e0)
Mar 23 13:29:13 escapepod start.sh[1401]:         /home/pi/wire-pod/chipper/pkg/wirepod/preqs/intent_graph.go:27 +0x53c
Mar 23 13:29:13 escapepod start.sh[1401]: github.com/kercre123/chipper/pkg/servers/chipper.(*Server).StreamingIntentGraph(0x4000206c40, {0x14467b8, 0x40005ad8e0})
Mar 23 13:29:13 escapepod start.sh[1401]:         /home/pi/wire-pod/chipper/pkg/servers/chipper/intent_graph.go:23 +0x290
Mar 23 13:29:13 escapepod start.sh[1401]: github.com/digital-dream-labs/api/go/chipperpb._ChipperGrpc_StreamingIntentGraph_Handler({0x121f620?, 0x4000206c40}, {0x1444c28?, 0x40004202a0})
Mar 23 13:29:13 escapepod start.sh[1401]:         /root/go/pkg/mod/github.com/digital-dream-labs/api@v0.0.0-20210824232136-8cc90c1bb12c/go/chipperpb/chipperpb_grpc.pb.go:294 +0x9c
Mar 23 13:29:13 escapepod start.sh[1401]: google.golang.org/grpc.(*Server).processStreamingRPC(0x4000142000, {0x1446fb8, 0x40004831e0}, 0x400016cc60, 0x4000226a80, 0x1a7d360, 0x0)
Mar 23 13:29:13 escapepod start.sh[1401]:         /root/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1629 +0xf44
Mar 23 13:29:13 escapepod start.sh[1401]: google.golang.org/grpc.(*Server).handleStream(0x4000142000, {0x1446fb8, 0x40004831e0}, 0x400016cc60, 0x0)
Mar 23 13:29:13 escapepod start.sh[1401]:         /root/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1717 +0x818
Mar 23 13:29:13 escapepod start.sh[1401]: google.golang.org/grpc.(*Server).serveStreams.func1.2()
Mar 23 13:29:13 escapepod start.sh[1401]:         /root/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:965 +0x84
Mar 23 13:29:13 escapepod start.sh[1401]: created by google.golang.org/grpc.(*Server).serveStreams.func1
Mar 23 13:29:13 escapepod start.sh[1401]:         /root/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:963 +0x294

**Comment by BillMerryman** [comment]
Installed a fresh image on the Pi and installed wire-pod again from scratch. Built the sdkTest plugin unchanged. When I try 'hey vector, hello world' I get:

Mar 23 22:10:33 escapepod start.sh[1467]: Text matched plugin SDK Plugin Test, executing function
Mar 23 22:10:33 escapepod start.sh[1467]: hello world plugin test
Mar 23 22:10:33 escapepod start.sh[1467]: 2023/03/23 22:10:33 vector-go-sdk error: Error opening chipper/jdocs/botSdkInfo.json, likely doesn't exist
Mar 23 22:10:33 escapepod start.sh[1467]: 2023/03/23 22:10:33 open chipper/jdocs/botSdkInfo.json: not a directory
Mar 23 22:10:33 escapepod start.sh[1467]: panic: runtime error: invalid memory address or nil pointer dereference
Mar 23 22:10:33 escapepod start.sh[1467]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7f464ef388]
Mar 23 22:10:33 escapepod start.sh[1467]: goroutine 103 [running]:
Mar 23 22:10:33 escapepod start.sh[1467]: github.com/fforchino/vector-go-sdk/pkg/vector.(*Vector).BehaviorControl(0x0?, {0x1442120?, 0x4000044098?}, 0x0?, 0x0?)
Mar 23 22:10:33 escapepod start.sh[1467]:         /root/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20221216080054-dfe1a258d335/pkg/vector/behaviorcontrol.go:13 +0x28
Mar 23 22:10:33 escapepod start.sh[1467]: plugin/unnamed-643af31f6039b5aa2f590bc76e01948ec703797e.Action.func1()
Mar 23 22:10:33 escapepod start.sh[1467]:         /home/pi/wire-pod/chipper/plugins/sdkTest.go:23 +0x38
Mar 23 22:10:33 escapepod start.sh[1467]: created by plugin/unnamed-643af31f6039b5aa2f590bc76e01948ec703797e.Action
Mar 23 22:10:33 escapepod start.sh[1467]:         /home/pi/wire-pod/chipper/plugins/sdkTest.go:22 +0x120

Which is strange because botSdkInfo.json is in my jdocs directory. When I modify the sdkTest example to subtitute 'InitSDK' for 'InitSDKForWirepod', I get:

Mar 23 22:17:33 escapepod start.sh[2093]: hello world plugin test
Mar 23 22:17:33 escapepod start.sh[2093]: 005063ac
Mar 23 22:17:33 escapepod start.sh[2093]: 192.168.2.20:443
Mar 23 22:17:33 escapepod start.sh[2093]: jb5vTn7MS6eMkq5RQpJvwQ==
Mar 23 22:17:33 escapepod start.sh[2093]: /home/pi/wire-pod/chipper/.anki_vector/Vector-D2F4-005063ac.cert
Mar 23 22:17:33 escapepod start.sh[2093]: Vector-D2F4
Mar 23 22:17:33 escapepod start.sh[2093]: panic: interface conversion: interface {} is nil, not string
Mar 23 22:17:33 escapepod start.sh[2093]: goroutine 43 [running]:
Mar 23 22:17:33 escapepod start.sh[2093]: github.com/fforchino/vector-go-sdk/pkg/sdk-wrapper.GetLocale(...)
Mar 23 22:17:33 escapepod start.sh[2093]:         /root/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20221216080054-dfe1a258d335/pkg/sdk-wrapper/sdk-wrapper-settings.go:127
Mar 23 22:17:33 escapepod start.sh[2093]: github.com/fforchino/vector-go-sdk/pkg/sdk-wrapper.refreshLanguage()
Mar 23 22:17:33 escapepod start.sh[2093]:         /root/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20221216080054-dfe1a258d335/pkg/sdk-wrapper/sdk-wrapper-voice.go:147 +0xe2c
Mar 23 22:17:33 escapepod start.sh[2093]: github.com/fforchino/vector-go-sdk/pkg/sdk-wrapper.RefreshSDKSettings()
Mar 23 22:17:33 escapepod start.sh[2093]:         /root/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20221216080054-dfe1a258d335/pkg/sdk-wrapper/sdk-wrapper-settings.go:52 +0xc8
Mar 23 22:17:33 escapepod start.sh[2093]: github.com/fforchino/vector-go-sdk/pkg/sdk-wrapper.InitSDK({0x4000044640?, 0x40000c8008?})
Mar 23 22:17:33 escapepod start.sh[2093]:         /root/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20221216080054-dfe1a258d335/pkg/sdk-wrapper/sdk-wrapper.go:54 +0x4a4
Mar 23 22:17:33 escapepod start.sh[2093]: plugin/unnamed-2724764ea56c68990762bcca8dcd671b059c2a09.Action({0x143b680?, 0x40000c8008?}, {0x4000044640, 0x8})
Mar 23 22:17:33 escapepod start.sh[2093]:         /home/pi/wire-pod/chipper/plugins/sdkTest.go:18 +0x6c
Mar 23 22:17:33 escapepod start.sh[2093]: github.com/kercre123/chipper/pkg/wirepod/ttr.pluginFunctionHandler({0x1108160, 0x4000474770}, {0x40004e66c0, 0xb}, 0x0?, {0x4000044640, 0x8})
Mar 23 22:17:33 escapepod start.sh[2093]:         /home/pi/wire-pod/chipper/pkg/wirepod/ttr/matchIntentSend.go:179 +0x188
Mar 23 22:17:33 escapepod start.sh[2093]: github.com/kercre123/chipper/pkg/wirepod/ttr.ProcessTextAll({0x1108160?, 0x4000474770?}, {0x40004e66c0?, 0x10?}, {0x40001fe600, 0x36, 0x480?}, {0x4000090c00?, 0x36, 0x40}, ...)
Mar 23 22:17:33 escapepod start.sh[2093]:         /home/pi/wire-pod/chipper/pkg/wirepod/ttr/matchIntentSend.go:214 +0x12c
Mar 23 22:17:33 escapepod start.sh[2093]: github.com/kercre123/chipper/pkg/wirepod/preqs.(*Server).ProcessIntentGraph(0x1448220?, 0x4000474770)
Mar 23 22:17:33 escapepod start.sh[2093]:         /home/pi/wire-pod/chipper/pkg/wirepod/preqs/intent_graph.go:27 +0x53c
Mar 23 22:17:33 escapepod start.sh[2093]: github.com/kercre123/chipper/pkg/servers/chipper.(*Server).StreamingIntentGraph(0x4000472180, {0x14467b8, 0x4000186190})
Mar 23 22:17:33 escapepod start.sh[2093]:         /home/pi/wire-pod/chipper/pkg/servers/chipper/intent_graph.go:23 +0x290
Mar 23 22:17:33 escapepod start.sh[2093]: github.com/digital-dream-labs/api/go/chipperpb._ChipperGrpc_StreamingIntentGraph_Handler({0x121f620?, 0x4000472180}, {0x1444c28?, 0x40006321c0})
Mar 23 22:17:33 escapepod start.sh[2093]:         /root/go/pkg/mod/github.com/digital-dream-labs/api@v0.0.0-20210824232136-8cc90c1bb12c/go/chipperpb/chipperpb_grpc.pb.go:294 +0x9c
Mar 23 22:17:33 escapepod start.sh[2093]: google.golang.org/grpc.(*Server).processStreamingRPC(0x400016e000, {0x1446fb8, 0x4000272820}, 0x4000000120, 0x40005142a0, 0x1a7d360, 0x0)
Mar 23 22:17:33 escapepod start.sh[2093]:         /root/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1629 +0xf44
Mar 23 22:17:33 escapepod start.sh[2093]: google.golang.org/grpc.(*Server).handleStream(0x400016e000, {0x1446fb8, 0x4000272820}, 0x4000000120, 0x0)
Mar 23 22:17:33 escapepod start.sh[2093]:         /root/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1717 +0x818
Mar 23 22:17:33 escapepod start.sh[2093]: google.golang.org/grpc.(*Server).serveStreams.func1.2()
Mar 23 22:17:33 escapepod start.sh[2093]:         /root/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:965 +0x84
Mar 23 22:17:33 escapepod start.sh[2093]: created by google.golang.org/grpc.(*Server).serveStreams.func1
Mar 23 22:17:33 escapepod start.sh[2093]:         /root/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:963 +0x294

**Comment by BillMerryman** [comment]
Ok, more intrigue. I created a separate directory off of home and set up one of the SDK examples (https://github.com/fforchino/vector-go-sdk/blob/main/cmd/examples/speak/main.go). I also installed a copy of my .anki_vector folder in there as well (I already have copies in my wire-pod folder, and my chipper folder, with the cert path appropriately updated in sdk_config.ini in both locations). When I ran the example, I was getting:

005063ac
192.168.2.20:443
jb5vTn7MS6eMkq5RQpJvwQ==
/home/pi/.anki_vector/Vector-D2F4-005063ac.cert
Vector-D2F4
2023/03/23 22:25:51 ERROR: Could not load Vector settings from JDOCS

I updated the example to use 'InitSDKForWirepod' instead of 'InitSDK' and got:

2023/03/23 22:29:06 vector-go-sdk error: Error opening chipper/jdocs/botSdkInfo.json, likely doesn't exist
2023/03/23 22:29:06 open chipper/jdocs/botSdkInfo.json: no such file or directory
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x974d28]

goroutine 7 [running]:
github.com/fforchino/vector-go-sdk/pkg/vector.(*Vector).BehaviorControl(0x0?, {0xc139f0?, 0x40000aa000?}, 0x0?, 0x0?)
	/home/pi/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20230322080941-ce71ec832247/pkg/vector/behaviorcontrol.go:13 +0x28
main.main.func1()
	/home/pi/vector-go-sdk-dev/main.go:20 +0x38
created by main.main
	/home/pi/vector-go-sdk-dev/main.go:19 +0x168
exit status 2

So I made a chipper/jdocs folder path in the directory for this SDK example and copied in the botSDKInfo.json and jdocs.json files and ran it again, AND IT RAN! But it also complained:

2023/03/23 22:30:53 Creating EP connection with robot on address 192.168.2.20:443, serialNo 005063ac, GUID tni1TRsTRTaNSapjo0Y+Sw==
Completing init with paths TMP:/tmp/, DATA:data, NVM:nvm
2023/03/23 22:30:54 WARNING: Could not load Vector custom settings, creating a blank file
2023/03/23 22:30:59 WARNING: Could not load Vector custom settings, creating a blank file
2023/03/23 22:31:00 WARNING: Could not load Vector custom settings, creating a blank file
ESPEAK ERROR exit status 127
Will save file /tmp/TTS-005063ac.wav
Request url: https://www.wondergarden.app/voiceserver/index.php/getText?text=Hello+world.+I+really+want+to+explore%21&lang=en-US&voice=TM:pqv7261xqhjk
filename="SGVsbG8gd29ybGQuIEkgcmVhbGx5IHdhbnQgdG8gZXhwbG9yZSE=.wav"
Name from web: SGVsbG8gd29ybGQuIEkgcmVhbGx5IHdhbnQgdG8gZXhwbG9yZSE=.wav"
Will save file /tmp/TTS-005063ac.wav
Request url: https://www.wondergarden.app/voiceserver/index.php/getText?text=Hello+world.+I+really+want+to+explore%21&lang=en-US&voice=TM:nczpqg58jj6n
filename="SGVsbG8gd29ybGQuIEkgcmVhbGx5IHdhbnQgdG8gZXhwbG9yZSE=.wav"
Name from web: SGVsbG8gd29ybGQuIEkgcmVhbGx5IHdhbnQgdG8gZXhwbG9yZSE=.wav"
Will save file /tmp/TTS-005063ac.wav
Request url: https://www.wondergarden.app/voiceserver/index.php/getText?text=Hello+world.+I+really+want+to+explore%21&lang=en-US&voice=TM:x8kck28kthq7
filename="SGVsbG8gd29ybGQuIEkgcmVhbGx5IHdhbnQgdG8gZXhwbG9yZSE=.wav"
Name from web: SGVsbG8gd29ybGQuIEkgcmVhbGx5IHdhbnQgdG8gZXhwbG9yZSE=.wav"
2023/03/23 22:31:22 WARNING: Could not load Vector custom settings, creating a blank file
2023/03/23 22:31:22 WARNING: Could not load Vector custom settings, creating a blank file
ESPEAK ERROR exit status 127
Will save file /tmp/TTS-005063ac.wav
Request url: https://www.wondergarden.app/voiceserver/index.php/getText?text=Ciao+mondo%21+Sono+pronto+per+esplorare&lang=it-IT&voice=TM:rdejatdh1r6h
filename="Q2lhbyBtb25kbyEgU29ubyBwcm9udG8gcGVyIGVzcGxvcmFyZQ==.wav"
Name from web: Q2lhbyBtb25kbyEgU29ubyBwcm9udG8gcGVyIGVzcGxvcmFyZQ==.wav"
2023/03/23 22:31:31 WARNING: Could not load Vector custom settings, creating a blank file
2023/03/23 22:31:31 WARNING: Could not load Vector custom settings, creating a blank file
ESPEAK ERROR exit status 127
Will save file /tmp/TTS-005063ac.wav
Request url: https://www.wondergarden.app/voiceserver/index.php/getText?text=Hallo+Welt%21+Ich+bin+bereit+zu+erkunden&lang=de-DE&voice=TM:pdp9dhyra0ph
filename="SGFsbG8gV2VsdCEgSWNoIGJpbiBiZXJlaXQgenUgZXJrdW5kZW4=.wav"
Name from web: SGFsbG8gV2VsdCEgSWNoIGJpbiBiZXJlaXQgenUgZXJrdW5kZW4=.wav"
2023/03/23 22:31:39 WARNING: Could not load Vector custom settings, creating a blank file
2023/03/23 22:31:39 WARNING: Could not load Vector custom settings, creating a blank file
ESPEAK ERROR exit status 127
Will save file /tmp/TTS-005063ac.wav
Request url: https://www.wondergarden.app/voiceserver/index.php/getText?text=Bonjour+le+monde%21+Je+suis+pr%C3%AAt+%C3%A0+explorer&lang=fr-FR&voice=TM:xn0ps57a5akf
filename="Qm9uam91ciBsZSBtb25kZSEgSmUgc3VpcyBwcsOqdCDDoCBleHBsb3Jlcg==.wav"
Name from web: Qm9uam91ciBsZSBtb25kZSEgSmUgc3VpcyBwcsOqdCDDoCBleHBsb3Jlcg==.wav"
2023/03/23 22:31:48 WARNING: Could not load Vector custom settings, creating a blank file
2023/03/23 22:31:48 WARNING: Could not load Vector custom settings, creating a blank file
ESPEAK ERROR exit status 127
Will save file /tmp/TTS-005063ac.wav
Request url: https://www.wondergarden.app/voiceserver/index.php/getText?text=%C2%A1Hola+Mundo%21+Estoy+listo+para+explorar&lang=es-ES&voice=TM:09awctyevpey
filename="wqFIb2xhIE11bmRvISBFc3RveSBsaXN0byBwYXJhIGV4cGxvcmFy.wav"
Name from web: wqFIb2xhIE11bmRvISBFc3RveSBsaXN0byBwYXJhIGV4cGxvcmFy.wav"

**Comment by BillMerryman** [comment]
Not really sure what brought the issue on, but after completely wiping Vector (including moving back to DDL servos), doing a fresh OS install on the Pi, re-setting up the Python SDK (this time from the Pi), and re-onboarding Vector, it appears my plugin is working again, as well as everything else. Setting up the Python SDK on the Pi places the .anki_vector folder in the home directory, but the Go SDK looks for it in the chipper folder, so placed a copy of it there and updated the sdk_config.ini file so the path to the cert was correct. Anyway, it's working now. I realize I didn't copy my code into this anywhere, so will place it here now as part of closing this issue. You will need to setup a Clarifai account, and replace the '{your Clarifai api key here}' with your API key. When running, asking Vector "What is this thing", he will capture an image of it, submit it to Clarifai, and he will respond by telling you the first five categories of things they identify it as:

package main

import (
	sdk_wrapper "github.com/fforchino/vector-go-sdk/pkg/sdk-wrapper"
	"context"
	"fmt"
	"github.com/Clarifai/clarifai-go-grpc/proto/clarifai/api"
	"github.com/Clarifai/clarifai-go-grpc/proto/clarifai/api/status"
	"google.golang.org/grpc"
	"google.golang.org/grpc/credentials"
	"google.golang.org/grpc/metadata"
	"io/ioutil"
	"strings"
)

var Utterances = []string{"what is this thing"}
var Name = "Identifier Plugin"

func Action(transcribedText string, botSerial string) string {

	sdk_wrapper.InitSDK(botSerial)
	ctx := context.Background()
	start := make(chan bool)
	stop := make(chan bool)

	go func() {
		_ = sdk_wrapper.Robot.BehaviorControl(ctx, start, stop)
	}()

	for {
		select {
		case <-start:
			sdk_wrapper.SaveHiResCameraPicture("object.jpg")
			
			conn, err := grpc.Dial(
				"api.clarifai.com:443",
				grpc.WithTransportCredentials(credentials.NewClientTLSFromCert(nil, "")),
			)
			if err != nil {
				panic(err)
			}
			client := api.NewV2Client(conn)

			ctx_Clarifai := metadata.AppendToOutgoingContext(
				context.Background(),
				"Authorization", "Key {your Clarifai api key here}",
			)

			var GeneralModelId = "aaa03c23b3724a16a56b629203edc62c"

			fileBytes, err := ioutil.ReadFile("object.jpg")
			
			response, err := client.PostModelOutputs(
				ctx_Clarifai,
				&api.PostModelOutputsRequest{
				ModelId: GeneralModelId,
				Inputs: []*api.Input{
					{
						Data: &api.Data{
							Image: &api.Image{
								Base64: fileBytes,
							},
						},
					},
				},
				},
			)
			
			if err != nil {
				panic(err)
			}
			
			if response.Status.Code != status.StatusCode_SUCCESS {
				panic(fmt.Sprintf("Failed response: %s", response))
			}
			
			//Get the object name with response.Outputs[0].Data.Concepts[0].Name
			//Get the likelihood with response.Outputs[0].Data.Concepts[0].Value
			sdk_wrapper.SayText("That, my friend, could be any of the following things.")
			for index := 0; index < 5; index++ {
				var Article string
				Name := response.Outputs[0].Data.Concepts[index].Name
				if strings.Contains("AEIOUaeiou", Name[0:1]){
					Article = "an"
				}else{
					Article = "a"
				}
				sdk_wrapper.SayText(Article + " " + Name)
			}
			stop <- true
			break
		}
	}

	return "intent_imperative_praise"

}

**Comment by BillMerryman** [comment]
Guess I spoke too soon. Was working fine, even after rebooting the server and sleeping and waking Vector. I don't leave my Vectors running all the time, so I put the one I have on Wirepod away for a week. I left the Pi server running, without making any changes (not even updates). Came back today and fired up Vector, asked him 'What is this thing', and got this:

Apr 13 13:41:16 escapepod start.sh[2513]: panic: interface conversion: interface {} is nil, not string
Apr 13 13:41:16 escapepod start.sh[2513]: goroutine 39 [running]:
Apr 13 13:41:16 escapepod start.sh[2513]: github.com/fforchino/vector-go-sdk/pkg/sdk-wrapper.GetLocale(...)
Apr 13 13:41:16 escapepod start.sh[2513]:         /root/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20221216080054-dfe1a258d335/pkg/sdk-wrapper/sdk-wrapper-settings.go:127
Apr 13 13:41:16 escapepod start.sh[2513]: github.com/fforchino/vector-go-sdk/pkg/sdk-wrapper.refreshLanguage()
Apr 13 13:41:16 escapepod start.sh[2513]:         /root/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20221216080054-dfe1a258d335/pkg/sdk-wrapper/sdk-wrapper-voice.go:147 +0xe2c
Apr 13 13:41:16 escapepod start.sh[2513]: github.com/fforchino/vector-go-sdk/pkg/sdk-wrapper.RefreshSDKSettings()
Apr 13 13:41:16 escapepod start.sh[2513]:         /root/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20221216080054-dfe1a258d335/pkg/sdk-wrapper/sdk-wrapper-settings.go:52 +0xc8
Apr 13 13:41:16 escapepod start.sh[2513]: github.com/fforchino/vector-go-sdk/pkg/sdk-wrapper.InitSDK({0x400048a278?, 0x0?})
Apr 13 13:41:16 escapepod start.sh[2513]:         /root/go/pkg/mod/github.com/fforchino/vector-go-sdk@v0.0.0-20221216080054-dfe1a258d335/pkg/sdk-wrapper/sdk-wrapper.go:54 +0x4a4
Apr 13 13:41:16 escapepod start.sh[2513]: plugin/unnamed-e5b0aa080af34a0a94150f2d9b7255e14ddb467c.Action({0x143b680?, 0x40000c8008?}, {0x400048a278?, 0x1?})
Apr 13 13:41:16 escapepod start.sh[2513]:         /home/pi/wire-pod/chipper/plugins/identifier.go:21 +0x30
Apr 13 13:41:16 escapepod start.sh[2513]: github.com/kercre123/chipper/pkg/wirepod/ttr.pluginFunctionHandler({0x1108160, 0x40001ec8c0}, {0x4000478060, 0x12}, 0x0?, {0x400048a278, 0x8})
Apr 13 13:41:16 escapepod start.sh[2513]:         /home/pi/wire-pod/chipper/pkg/wirepod/ttr/matchIntentSend.go:179 +0x188
Apr 13 13:41:16 escapepod start.sh[2513]: github.com/kercre123/chipper/pkg/wirepod/ttr.ProcessTextAll({0x1108160?, 0x40001ec8c0?}, {0x4000478060?, 0x10?}, {0x40001fe600, 0x36, 0x400?}, {0x4000090c00?, 0x36, 0x40}, ...)
Apr 13 13:41:16 escapepod start.sh[2513]:         /home/pi/wire-pod/chipper/pkg/wirepod/ttr/matchIntentSend.go:214 +0x12c
Apr 13 13:41:16 escapepod start.sh[2513]: github.com/kercre123/chipper/pkg/wirepod/preqs.(*Server).ProcessIntentGraph(0x1448220?, 0x40001ec8c0)
Apr 13 13:41:16 escapepod start.sh[2513]:         /home/pi/wire-pod/chipper/pkg/wirepod/preqs/intent_graph.go:27 +0x53c
Apr 13 13:41:16 escapepod start.sh[2513]: github.com/kercre123/chipper/pkg/servers/chipper.(*Server).StreamingIntentGraph(0x40002ac600, {0x14467b8, 0x40001aef80})
Apr 13 13:41:16 escapepod start.sh[2513]:         /home/pi/wire-pod/chipper/pkg/servers/chipper/intent_graph.go:23 +0x290
Apr 13 13:41:16 escapepod start.sh[2513]: github.com/digital-dream-labs/api/go/chipperpb._ChipperGrpc_StreamingIntentGraph_Handler({0x121f620?, 0x40002ac600}, {0x1444c28?, 0x400025e380})
Apr 13 13:41:16 escapepod start.sh[2513]:         /root/go/pkg/mod/github.com/digital-dream-labs/api@v0.0.0-20210824232136-8cc90c1bb12c/go/chipperpb/chipperpb_grpc.pb.go:294 +0x9c
Apr 13 13:41:16 escapepod start.sh[2513]: google.golang.org/grpc.(*Server).processStreamingRPC(0x40002e0000, {0x1446fb8, 0x40004d04e0}, 0x40000dca20, 0x40002a3770, 0x1a7d360, 0x0)
Apr 13 13:41:16 escapepod start.sh[2513]:         /root/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1629 +0xf44
Apr 13 13:41:16 escapepod start.sh[2513]: google.golang.org/grpc.(*Server).handleStream(0x40002e0000, {0x1446fb8, 0x40004d04e0}, 0x40000dca20, 0x0)
Apr 13 13:41:16 escapepod start.sh[2513]:         /root/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1717 +0x818
Apr 13 13:41:16 escapepod start.sh[2513]: google.golang.org/grpc.(*Server).serveStreams.func1.2()
Apr 13 13:41:16 escapepod start.sh[2513]:         /root/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:965 +0x84
Apr 13 13:41:16 escapepod start.sh[2513]: created by google.golang.org/grpc.(*Server).serveStreams.func1
Apr 13 13:41:16 escapepod start.sh[2513]:         /root/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:963 +0x294

Same error I was getting before. Don't know why it starts doing this after some amount of time.

**Comment by kercre123** [comment]
I have sort of given up on plugins at this point because they compile to relatively huge sizes and there is already a "custom intent" system which can launch external applications, like Python scripts (or compiled Go binaries, if you want speed).

I got it working reliably with the digital-dream-labs/vector-go-sdk, but it involved modifying the plugin code so it launches the Action function with the bot IP and token, and I had to work around the gRPC definition errors with an env var. I may push that code at some point, but it doesn't seem like a great experience to me.

**Comment by BillMerryman** [comment]
I think I discovered where the issue is coming from. I reset everything and got wire-pod and my plugin re-set up (including running anki_vector.configure). I then stepped away from it for a couple of days. When I returned, the plugin was once again (as I expected) failing. This time I re-ran anki_vector.configure and replaced the existing .anki_vector folder, and the plugin started working again! So I'm guessing there is some kind of issue with the certificate DDL is providing through anki_vector.configure. I'm going to look for something to parse the certificate to see if there is anything suspect.
